lib/beautiful/init.lua

```diff
@@ -111,22 +111,34 @@ end
 function beautiful.init(config)
     if config then
         local homedir = os.getenv("HOME")
+        local theme_path = ""
 
         -- If `config` is the path to a theme file, run this file,
         -- otherwise if it is a theme table, save it.
         if type(config) == 'string' then
             -- Expand the '~' $HOME shortcut
             config = config:gsub("^~/", homedir .. "/")
             theme = protected_call(dofile, config)
+            theme_path = config:gsub("/[^/]+$", "")
         elseif type(config) == 'table' then
             theme = config
         end
 
         if theme then
-            -- expand '~'
             if homedir then
-                for k, v in pairs(theme) do
-                    if type(v) == "string" then theme[k] = v:gsub("^~/", homedir .. "/") end
+                -- expand '~' and './'
+                if type(config) == "string" then
+                    for k, v in pairs(theme) do
+                        if type(v) == "string" then
+                            theme[k] = v:gsub("^~/", homedir .. "/")
+                            theme[k] = v:gsub("^%./", theme_path .. "/")
+                        end
+                    end
+                -- expand '~'
+                elseif type(config) == 'table' then
+                    for k, v in pairs(theme) do
+                        if type(v) == "string" then theme[k] = v:gsub("^~/", homedir .. "/") end
```

Comment:
In this case, the `config` is a table instead of a file path, it cannot decide the base path unless specify one explicitly in the table.